server/db/postgres/adapter.go

```diff
@@ -0,0 +1,2696 @@
+	}
+	return err
+}
+
+// IsOpen returns true if connection to database has been established. It does not check if
+// connection is actually live.
+func (a *adapter) IsOpen() bool {
+	fmt.Println("inside adapter isOpen()")
+	return a.db != nil
+}
+
+// GetDbVersion returns current database version.
+func (a *adapter) GetDbVersion() (int, error) {
+	if a.version > 0 {
+		return a.version, nil
+	}
+
+	var vers int
+	err := a.db.Get(&vers, "SELECT `value` FROM kvmeta WHERE `key`='version'")
+	if err != nil {
+		if isMissingDb(err) || err == sql.ErrNoRows {
+			err = errors.New("Database not initialized")
+		}
+		return -1, err
+	}
+
+	a.version = vers
+
+	return vers, nil
+}
+
+func (a *adapter) updateDbVersion(v int) error {
+	a.version = -1
+	if _, err := a.db.Exec("UPDATE kvmeta SET `value`=? WHERE `key`='version'", v); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CheckDbVersion checks whether the actual DB version matches the expected version of this adapter.
+func (a *adapter) CheckDbVersion() error {
+	/*
+		version, err := a.GetDbVersion()
+		if err != nil {
+			return err
+		}
+
+		if version != adpVersion {
+			return errors.New("Invalid database version " + strconv.Itoa(version) +
+				". Expected " + strconv.Itoa(adpVersion))
+		}
+	*/
+	return nil
+}
+
+// Version returns adapter version.
+func (adapter) Version() int {
+	return adpVersion
+}
+
+// GetName returns string that adapter uses to register itself with store.
+func (a *adapter) GetName() string {
+	return adapterName
+}
+
+// SetMaxResults configures how many results can be returned in a single DB call.
+func (a *adapter) SetMaxResults(val int) error {
+	if val <= 0 {
+		a.maxResults = defaultMaxResults
+	} else {
+		a.maxResults = val
+	}
+
+	return nil
+}
+
+// CreateDb initializes the storage.
+func (a *adapter) CreateDb(reset bool) error {
+	var err error
+	var tx *sql.Tx
+
+	// Can't use an existing connection because it's configured with a database name which may not exist.
+	// Don't care if it does not close cleanly.
+	a.db.Close()
+
+	// This DSN has been parsed before and produced no error, not checking for errors here.
+
+	fmt.Println("opening new connection in createdb")
+	a.db, err = sqlx.Connect("postgres", a.dsn)
+	fmt.Println(a.db, err)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			tx.Rollback()
+		}
+	}()
+
+	fmt.Println("droping db")
+	result := a.db.MustExec("DROP DATABASE IF EXISTS " + a.dbName)
+	fmt.Println("dropped db")
+	fmt.Println(result)
+
+	result = a.db.MustExec("CREATE DATABASE " + a.dbName)
+	fmt.Println("created db")
+	fmt.Println(result)
+	a.db.Close()
+
+	dsn := strings.Replace(a.dsn, "?", "/"+a.dbName+"?", 1)
+	a.db, err = sqlx.Connect("postgres", dsn)
+	if err != nil {
+		return err
+	}
+	//result = a.db.MustExec("\\c" + a.dbName)
+
+	if tx, err = a.db.Begin(); err != nil {
+		return err
+	}
+	fmt.Println("connected")
+	fmt.Println(result)
+
+	if _, err = tx.Exec(
+		`CREATE TABLE users(
```

Comment:
Instead of executing these "create table" statements separately,
would it be easier to keep the schema separately and apply it in one go?
Check the example in https://github.com/jmoiron/sqlx#usage